acceptance/tests/agent/last_run_summary_report.rb

```diff
@@ -0,0 +1,108 @@
+test_name "The 'last_run_summary.yaml' report has the right location and permissions" do
+  tag 'risk:medium',
+      'audit:high'
+
+  require 'puppet/acceptance/temp_file_utils'
+  extend Puppet::Acceptance::TempFileUtils
+  
+  agents.each do |agent|
+    custom_publicdir = agent.tmpdir('custom_public_dir')
+
+    statedir = on(agent, puppet('config print statedir')).stdout.chomp
+    fail_test("The 'statedir' config is not set!") if statedir.empty?
+
+    publicdir = on(agent, puppet('config print publicdir')).stdout.chomp
+    fail_test("The 'publicdir' config is not set!") if publicdir.empty?
+
+    teardown do
+      on(agent, "rm -rf #{custom_publicdir}/*")
+      on(agent, "rm -rf #{publicdir}/*") unless publicdir.empty?
```

Comment:
Not mandatory, but beaker has a `rm_rf` helper. Also, the globs aren't necessary if you want to remove the entire directory (which you probably want to do for `custom_publicdir`).

```suggestion
      agent.rm_rf(custom_publicdir)
      agent.rm_rf("#{publicdir}/*") unless publicdir.empty?
```